filament/src/materials/ssao/sao.mat

```diff
@@ -287,18 +298,23 @@ fragment {
 
         float occlusion = 0.0;
         for (float i = 1.0; i <= materialParams.ssctRayCount.x; i += 1.0) {
-            cone.jitterOffset.x = random(gl_FragCoord.xy * i) * 2.0 - 1.0;      // direction
-            cone.jitterOffset.y = random(gl_FragCoord.xy * i * vec2(3, 11));    // step
+            cone.jitterOffset.x = random(getApiAgnosticFragCoords() * i) * 2.0 - 1.0;      // direction
+            cone.jitterOffset.y = random(getApiAgnosticFragCoords() * i * vec2(3, 11));    // step
             occlusion += coneTraceOcclusion(cone, materialParams_depth);
         }
         return occlusion * materialParams.ssctRayCount.y;
     }
 
+    // For SAO calculations, we make use of two coordinate spaces:
+    //     UV-space: framebuffer/texture space, graphics API dependent.
+    //     screen-space: origin bottom left, normalized coordinates.
+    // Textures must always be sampled with UV coordinates.
+
     void postProcess(inout PostProcessInputs postProcess) {
         highp vec2 uv = variable_vertex.xy; // interpolated to pixel center
 
         highp float depth = sampleDepthLinear(materialParams_depth, uv, 0.0, materialParams.depthParams);
```

Comment:
if you made `sampleDepthLinear` take a `ss` you'd be able to remove a lot of calls to `uvFromSs` -- might simplify things.

Reply:
Same thing with `computeViewSpaceNormal`.

I feel like we should do all the work in `ss` and pass `ss` everywhere, except at texture sampling time, where you can do a `uvFromSs`.